samples/datatype-sample/README.md

```diff
@@ -10,19 +10,100 @@ products:
 
 # Different data input sample for Kubernetes Marketplace Offer
 
-This sample demonstrates how to override array objects inside helm chart for Kubernetes Marketplace Offer
+This sample demonstrates 
+- how to override array objects inside helm chart for Kubernetes Marketplace Offer
+- how to override values on keys with special characters like '.' and inside deeper nested objects.
 
-In helm, there are 2 different ways to override array inside values.yaml
-- --set cpuLimit='{1,1.5}'
-- --set memoryLimit[0]=512Mi --set memoryLimit[1]=0.75Gi
+## Helm chart
 
-Only the second way can be used in Kubernetes Marketplace Offer. In this sample, we will show 2 different way to achieve generating the key value pairs that can be used to override array inside values.yaml
+To have azure-vote to consume the changes made in this sample, the following code snippet was added to values.yaml to allow user to override the CPU/Memory request and limit for the application.
+
+### values.yaml
+```yaml
+resourcesLimit:
+  # CPU request and limit array
+  cpuLimit:
+    - 0.25
+    - 0.5
+  # Memory request and limit array
+  memory.Limit:
+    - 128Mi
+    - 256Mi
+```
+
+Both CPU/Memory request and limit is nested under resourcesLimit. CPU request and limit is an array, while memory request and limit is also an array but with special character '.' in the key name.
+
+To consume these values properly from values.yaml, the following code snippet was added to deployments.yaml
+
+### deployments.yaml
+```yaml
+requests:
+cpu: {{ index .Values.resourcesLimit.cpuLimit 0}}
+memory: {{ index .Values "resourcesLimit" "memory.Limit" 0}}
+limits:
+cpu: {{ index .Values.resourcesLimit.cpuLimit 1}}
+memory: {{ index .Values "resourcesLimit" "memory.Limit" 1}}
+```
+In normal array, we can use index to access the value in the array. However, since memory.Limit has special character '.', we need to use index .Values "resourcesLimit" "memory.Limit" to access the value in the array.
+
+## How helm chart values are passed in through ARM template
+
+In helm, there are 2 different ways to override array inside values.yaml. Only the second way specified below can be used in Kubernetes Marketplace Offer.
+1. --set resourcesLimit.cpuLimit='{1,1.5}'
+2. --set resourcesLimit.cpuLimit[0]=1 --set resourcesLimit.cpuLimit[1]=1.5
+
+Only the second way can be used in Kubernetes Marketplace Offer. In this sample, we will show 2 different ways (editable grids, text boxes) to generate the key value pairs that can be used to override array inside values.yaml.
```

Comment:
nit: Text 'Only the second way can be used in Kubernetes Marketplace Offer' is redundant here. Please remove the text from this line.